packages/react-native/Libraries/Utilities/NativePlatformConstantsIOS.js

```diff
@@ -25,6 +25,7 @@ export interface Spec extends TurboModule {
     osVersion: string,
     systemName: string,
     interfaceIdiom: string,
+    isMacCatalyst: boolean,
```

Comment:
I'm not sure this approach is scalable.. 
we already have iOS, iPadOS and Catalyst. Soon, we will also have something for the xrOS in compatibility mode. 🤔 

What if we call this `applePlatform` and we return a string enum with `'iPhone' | 'iPad' | 'Catalyst' | 'xrOS' | 'unknown' as values?

Reply:
The difficulty is the idiom is either an iPad, or a Mac on Mac Catalyst - you get to pick

Reply:
I understand, but I don't get how that is related with the suggested change. Could you expand on that?
I'm suggsting a different way to represent the same information: we can populate the `applePlatform` variable with a similar logic, by using various `TARGET_OS_XXX` macros and then you can handle it in any way you want in JS, no?